README.md

```diff
@@ -124,6 +129,12 @@ resource "buildkite_pipeline" "repo2" {
 - `description` - (Optional) A description of the pipeline.
 - `repository` - (Required) The git URL of the repository.
 - `steps` - (Required) The string YAML steps to run the pipeline.
+- `team` - (Optional) Set team access for the pipeline. Can be specified multiple times for each team. Note that non-admin users may receive a buildkite permission error if trying to create a pipeline without at least one team. 
```

Comment:
To clarify how this works on the backend:

- If teams is not enabled on the organization (default for all new organizations)
  - There cannot be teams
- If teams is enabled on the organization (via https://buildkite.com/organizations/-/teams)
  - And the user is an admin
    - No teams are required
  - And the user is not an admin
    - And the user is only in a single team
      - No teams are required — it will default to the single team (usually the default "Everyone" team)
    - And the user is in multiple teams
      - Teams are required — otherwise there is ambiguity

Reply:
The simplest approach for the terraform provider is probably to make teams optional.

There is a small potential write/read inconsistency, though — if teams is enabled, the user is not an admin, and they are in a single team, and they do not provide teams, then the provider might read the resource after creation and see teams, and thus a diff.

A more complex approach would be to read the organization to see if teams is enabled and then dynamically decide if teams are required or not. I don't think we currently expose whether teams is enabled or not — I'll see if I can change that.

Reply:
> There is a small potential write/read inconsistency, though — if teams is enabled, the user is not an admin, and they are in a single team, and they do not provide teams, then the provider might read the resource after creation and see teams, and thus a diff.

In that scenario I don't think it would create the pipeline at all and would return`Error: Please ensure at least one team is added to this pipeline`. That was my experience anyway. And thinking more about it, that error is perfectly clear in this scenario and fixable now by the user as they can specify a team, so I don't think I need the disclaimer bit anymore.